api/routes.go

```diff
@@ -65,17 +80,26 @@ func (api *API) noValidate(h httprouter.Handle) httprouter.Handle {
 func (api *API) validate(h httprouter.Handle) httprouter.Handle {
 	return func(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
 		api.logRequest(req)
-		tokenStr, err := tokenFromRequest(req)
-		if err != nil {
-			api.staticLogger.Debugln("Error fetching token from request:", err)
-			api.WriteError(w, err, http.StatusUnauthorized)
-			return
-		}
-		token, err := jwt.ValidateToken(tokenStr)
-		if err != nil {
-			api.staticLogger.Debugln("Error validating token:", err)
-			api.WriteError(w, err, http.StatusUnauthorized)
-			return
+		var token jwt2.Token
+		// Check for an API key. We only return an error if an invalid API key
+		// is provided.
+		ak, err := apiKeyFromRequest(req)
+		if err == nil {
+			// We have an API key. Let's generate a token based on it.
+			token, err = api.tokenFromAPIKey(req.Context(), ak)
+			if err != nil {
+				api.staticLogger.Debugln("Error generating token for API key:", err)
```

Comment:
nit: in general I would have all errors logs as printlns so that we don't need to rebuild in dev mode to get error logs.

Reply:
We don't need to rebuild - there's an env var called `SKYNET_ACCOUNTS_LOG_LEVEL` which is read on startup and configures the logger. This allows me to put way more logging than generally needed and spread it across levels, so I can turn more and more of it until I see enough to be able to diagnose the problem without drowning in a sea of text.

If anything, I would like to make it more modular, so I can increase the log level in a specific module in order to diagnose what's happening there. I've used this approach in the past and it was amazing for quickly pinning down where the issue is.